pkg/driver/deployment/knative.go

```diff
@@ -46,15 +48,25 @@ func newKnativeDeployerConfiguration(cfg *config.Configuration) knativeDeploymen
 func (*knativeDeployer) Deploy(cfg *config.Configuration) {
 	knativeConfig := newKnativeDeployerConfiguration(cfg)
 
+	queue := make(chan struct{}, runtime.NumCPU()) // message queue as a sync method
+	deployed := sync.WaitGroup{}
+	deployed.Add(len(cfg.Functions))
+
 	for i := 0; i < len(cfg.Functions); i++ {
+		queue <- struct{}{}
+
 		knativeDeploySingleFunction(
```

Comment:
You missed the `go` in front of this call. The point is to do that in goroutines.

Reply:
Fixed in https://github.com/vhive-serverless/invitro/pull/557/commits/c54df798ee3910fcd72c2a4d6df34fa55a834e7c.

@leokondrashov: Can you double-check that the code is correct now?